File: microservice-gestion-prof/src/main/java/com/isi/service/ProfesseurService.java
package com.isi.service;

import com.isi.dto.ProfesseurDTO;
import com.isi.model.Professeur;
import org.springframework.data.domain.Page;

import java.util.List;

public interface ProfesseurService {
    Professeur createProfesseur(ProfesseurDTO professeurDTO);
    Professeur updateProfesseur(Long id, ProfesseurDTO professeurDTO);
    void deleteProfesseur(Long id);
    Professeur getProfesseurById(Long id);
    List<Professeur> getAllProfesseurs();
    Page<Professeur> getAllProfesseursPagines(int page, int size);
    List<Professeur> getProfesseursByDepartement(Long departementId);
    List<Professeur> getProfesseursBySpecialite(Long specialiteId);

}

File: frontenddevops/src/app/modules/professeur/services/professeur.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { Professeur, ProfesseurDTO } from '../models/professeur';

@Injectable({
  providedIn: 'root'
})
export class ProfesseurService {
  private apiUrl = environment.apiUrls.microserviceprofesseur + 'professeurs';
  constructor(private http: HttpClient) { }

  getAllProfesseurs(): Observable<Professeur[]> {
    return this.http.get<Professeur[]>(this.apiUrl);
  }

  getProfesseurById(id: number): Observable<Professeur> {
    return this.http.get<Professeur>(`${this.apiUrl}/${id}`);
  }

  createProfesseur(professeur: ProfesseurDTO): Observable<Professeur> {
    return this.http.post<Professeur>(this.apiUrl, professeur);
  }

  updateProfesseur(id: number, professeur: ProfesseurDTO): Observable<Professeur> {
    return this.http.put<Professeur>(`${this.apiUrl}/${id}`, professeur);
  }

  deleteProfesseur(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  getProfesseursByDepartement(departementId: number): Observable<Professeur[]> {
    return this.http.get<Professeur[]>(`${this.apiUrl}/departement/${departementId}`);
  }

  getProfesseursBySpecialite(specialiteId: number): Observable<Professeur[]> {
    return this.http.get<Professeur[]>(`${this.apiUrl}/specialite/${specialiteId}`);
  }
}